Eradication/ReportHIVByAgeAndGender.cpp

```diff
@@ -246,6 +273,58 @@ namespace Kernel
         return ret;
     }
 
+    void ReportHIVByAgeAndGender::CreateAdditionalRestrictionsList( const json::Element& rJsonElement,
+                                                                    const std::string& rDataLocation,
+                                                                    const char* parameterName,
+                                                                    std::vector<IAdditionalRestrictions*>& arList )
+    {
+        if( rJsonElement.Type() == json::NULL_ELEMENT )
+        {
+            std::stringstream ss;
+            ss << "'ReportHIVByAgeAndGender' found the element to be NULL for '" << parameterName << "' in <" << rDataLocation << ">.";
+            throw FactoryCreateFromJsonException( __FILE__, __LINE__, __FUNCTION__, ss.str().c_str() );
+        }
+
+        if( rJsonElement.Type() != json::ARRAY_ELEMENT )
+        {
+            std::stringstream ss;
+            ss << "'ReportHIVByAgeAndGender' found the element specified by '" << parameterName << "'\n"
+                << "to NOT be a JSON ARRAY in <" << rDataLocation << ">.";
+            throw FactoryCreateFromJsonException( __FILE__, __LINE__, __FUNCTION__, ss.str().c_str() );
+        }
+        const json::Array& ar_array = json::QuickInterpreter(rJsonElement).As<json::Array>();
+
+        if( ar_array.Size() == 0 )
+        {
+            std::stringstream ss;
+            ss << "'ReportHIVByAgeAndGender' found zero elements in JSON for '" << parameterName << "' in <" << rDataLocation << ">.";
+            throw FactoryCreateFromJsonException( __FILE__, __LINE__, __FUNCTION__, ss.str().c_str() );
+        }
+
+        for( int idx = 0; idx < ar_array.Size(); ++idx )
+        {
+            std::stringstream param_name;
+            param_name << parameterName << "[" << idx << "]";
+
+            const json::Element& r_array_element = ar_array[ idx ];
+            if( r_array_element.Type() != json::OBJECT_ELEMENT )
+            {
+                std::stringstream ss;
+                ss << "'ReportHIVByAgeAndGender' found the element specified by '" << param_name.str() << "'\n"
+                    << "to NOT be a JSON OBJECT in <" << rDataLocation << ">.";
```

Comment:
This error wouldn't be helpful to me. What is "not a json object". is "json object" a dictionary?

Reply:
I looked and there are at least two others of these errors in EMOD already. I think they are also not very helpful.

Reply:
FYI.  JSON = JavaScript Object Notation .  See https://www.w3schools.com/js/js_json_objects.asp

I think you want to call it a dictionary because Python dictionaries can look a lot like JSON objects.